### DISCUSSION TITLE
Cannot run GUPPY_SHAPERS as numpy is not present.

### DISCUSSION THREAD

**Discussion by Krytmn** [Guilouz/Creality-Helper-Script-Wiki#532] [Issues]
So, running input shaping after installing GuppyScreen isn't possible, for me anyway. The source of the issue seems to be that the numpy module isn't present. Of course pip install numpy doesn't work, because why would it? Console readout follows:

> // import numpy as np, matplotlib
> // ModuleNotFoundError: No module named 'numpy'
> // Command {guppy_input_shaper} finished
> // Running Command {guppy_input_shaper}...:
> // Traceback (most recent call last):
> // File "/usr/data/printer_data/config/GuppyScreen/scripts/calibrate_shaper.py", line 12, in
> // import numpy as np, matplotlib
> // ModuleNotFoundError: No module named 'numpy'
> // Command {guppy_input_shaper} finished

Any ideas as to how to remedy this would be fantastic. Thanks in advance!

**Comment by Guilouz** [comment]
Those error usually means the global python installation is screwed up. Have you done anything to the python installation in /usr/bin and libs in /usr/lib?

**Reply by Krytmn** [reply]
No, unfortunately. This is a fresh install. I was on a slightly outdated firmware. Completely reset the printer, flashed the latest firmware, went down the list installing the add-ons I enjoy, none of which threw errors, very odd. 

I suppose I should reset the printer again and just re-install?

**Reply by Guilouz** [reply]
Yes reset must be done after firmware update.

**Reply by Krytmn** [reply]
Hey, just for the sake of finality, resetting just presented a separate issue entirely. I did a reset and went straight to installing GuppyScreen after the reset. So after a fresh reset I installed only Klipper Gcode Shell Command and GuppyScreen, nothing else. After GuppyScreen installed, with no errors shown, the printer was stuck on "Waiting for printer to initialize."

Several hart restarts and no change, just forever stuck on "Waiting for printer to initialize." Fortunately SSH over wifi still worked so I was able to just reset again, but nothing else worked. I'm just skipping GuppyScreen entirely, doesn't seem to be my fate to use it. 

In any case, thank you for the work you put into this, it is incredibly helpful and I suggest it to people all of the time.

Have a great weekend!

Edit: I just realized I didn't install Moonraker after the reset. That was likely the issue and I reset for nothing >_<. I'll try again. Maybe that could be a required dependency for GuppyScreen just like the Gcode Shell Command, for dummies like me. 

Edit2: Actually, no luck, same issue, waiting for printer to initialize. I just uninstalled GuppyScreen via the helper script. Not sure if this would be expected but during the uninstall this appeared:

> Info: Restarting Moonraker service...
> start-stop-daemon: warning: killing process 19067: No such process

**Comment by Krytmn** [comment]
To be clear I did do a full reset via the tool both before and after
flashing the updated firmware but, that is what I'll do. Simple enough fix.
